packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2001,65 +2034,172 @@ export function writeEndSegment(
 //   }
 // }
 //
-// function completeBoundary(suspenseBoundaryID, contentID) {
-//   // Find the fallback's first element.
-//   const suspenseIdNode = document.getElementById(suspenseBoundaryID);
-//   const contentNode = document.getElementById(contentID);
-//   // We'll detach the content node so that regardless of what happens next we don't leave in the tree.
-//   // This might also help by not causing recalcing each time we move a child from here to the target.
-//   contentNode.parentNode.removeChild(contentNode);
-//   if (!suspenseIdNode) {
-//     // The user must have already navigated away from this tree.
-//     // E.g. because the parent was hydrated. That's fine there's nothing to do
-//     // but we have to make sure that we already deleted the container node.
-//     return;
+// const insertStyles = (function() {
+//   const resourceMap = new Map();
+//   const precedences = new Map();
+//   // We omit rel stylesheet because only stylesheets should have data-prec attribute
+//   // and we can concievably use another kind of link to act as a placeholder for a
+//   // precedence that does not yet have any resources.
+//   let lastResource, node;
+//   let thisDocument = document;
+
+//   // Seed the precedence list with existing resources
+//   let nodes = thisDocument.querySelectorAll('link[data-rprec]');
+//   for (let i = 0;node = nodes[i++];) {
+//     precedences.set(node.dataset.rprec, lastResource = node);
 //   }
-//   // Find the boundary around the fallback. This is always the previous node.
-//   const suspenseNode = suspenseIdNode.previousSibling;
-//
-//   // Clear all the existing children. This is complicated because
-//   // there can be embedded Suspense boundaries in the fallback.
-//   // This is similar to clearSuspenseBoundary in ReactDOMHostConfig.
-//   // TODO: We could avoid this if we never emitted suspense boundaries in fallback trees.
-//   // They never hydrate anyway. However, currently we support incrementally loading the fallback.
-//   const parentInstance = suspenseNode.parentNode;
-//   let node = suspenseNode.nextSibling;
-//   let depth = 0;
-//   do {
-//     if (node && node.nodeType === COMMENT_NODE) {
-//       const data = node.data;
-//       if (data === SUSPENSE_END_DATA) {
-//         if (depth === 0) {
-//           break;
-//         } else {
-//           depth--;
+
+//   function getPromise(resourceEl) {
+//     let p = resourceEl._p = new Promise((re, rj) => {
+//       resourceEl.onload = re;
+//       resourceEl.onerror = rj;
+//     }).then(() => {
+//       p.s = 'l';
+//     }, (e) => {
+//       p.s = 'e';
+//       throw e;
```

Comment:
You can just stash the original promise instead of the chained one to avoid this.

Reply:
done